dds/DCPS/LinuxNetworkConfigPublisher.cpp

```diff
@@ -0,0 +1,243 @@
+      ACE_ERROR((LM_ERROR, "LinuxNetworkConfigPublisher::process_message: NETLINK error: %C\n", strerror(-msg->error)));
+    }
+    break;
+  case RTM_NEWADDR:
+    {
+      const struct ifaddrmsg* msg = reinterpret_cast<struct ifaddrmsg*>(NLMSG_DATA(header));
+      size_t address_length = 0;
+      switch(msg->ifa_family) {
+      case AF_INET:
+        address_length = 4;
+        break;
+      default:
+        ACE_ERROR((LM_WARNING, "LinuxNetworkConfigPublisher::process_message: unhandled address family: %d\n", msg->ifa_family));
+        return;
+      }
+      int rta_length = IFA_PAYLOAD(header);
+      for (const struct rtattr* attr = reinterpret_cast<const struct rtattr*>(IFA_RTA(msg));
+           RTA_OK(attr, rta_length);
+           attr = RTA_NEXT(attr, rta_length)) {
+        switch (attr->rta_type) {
+        case IFA_ADDRESS:
+          {
+            if (RTA_PAYLOAD(attr) != address_length) {
+              ACE_ERROR((LM_WARNING, "LinuxNetworkConfigPublisher::process_message: incorrect address byte count\n"));
+              return;
+            }
+            ACE_INET_Addr addr;
+            addr.set_address(reinterpret_cast<const char*>(RTA_DATA(attr)), address_length, 0);
+            add_address(msg->ifa_index, addr);
+          }
+          break;
+        }
+      }
+
+    }
+    break;
+  case RTM_DELADDR:
+    {
+      const struct ifaddrmsg* msg = reinterpret_cast<struct ifaddrmsg*>(NLMSG_DATA(header));
+      size_t address_length = 0;
+      switch(msg->ifa_family) {
+      case AF_INET:
+        address_length = 4;
+        break;
+      default:
+        ACE_ERROR((LM_WARNING, "LinuxNetworkConfigPublisher::process_message: unhandled address family: %d\n", msg->ifa_family));
+        return;
+      }
+      int rta_length = IFA_PAYLOAD(header);
+      for (const struct rtattr* attr = reinterpret_cast<const struct rtattr*>(IFA_RTA(msg));
+           RTA_OK(attr, rta_length);
+           attr = RTA_NEXT(attr, rta_length)) {
+        switch (attr->rta_type) {
+        case IFA_ADDRESS:
+          {
+            if (RTA_PAYLOAD(attr) != address_length) {
+              ACE_ERROR((LM_WARNING, "LinuxNetworkConfigPublisher::process_message: incorrect address byte count\n"));
+              return;
+            }
+            ACE_INET_Addr addr;
+            addr.set_address(reinterpret_cast<const char*>(RTA_DATA(attr)), address_length, 0);
+            remove_address(msg->ifa_index, addr);
+          }
+          break;
+        }
+      }
+
+    }
+    break;
+  case RTM_NEWLINK:
+    {
+      OPENDDS_STRING name;
+      const struct ifinfomsg* msg = reinterpret_cast<struct ifinfomsg*>(NLMSG_DATA(header));
+      int rta_length = IFLA_PAYLOAD(header);
+      for (const struct rtattr* attr = reinterpret_cast<const struct rtattr*>(IFLA_RTA(msg));
+           RTA_OK(attr, rta_length);
+           attr = RTA_NEXT(attr, rta_length)) {
+        switch (attr->rta_type) {
```

Comment:
```suggestion
        if (attr->rta_type == IFLA_IFNAME) {
```

Unless maybe you're planning on supporting more types.